### PR TITLE
[BugFix] Keep update statement output column's type consistent with target table's columns

### PR DESCRIPTION
**Pull request by LiShuMing**
## Why I'm doing:

- Update will fail when insert value is  constant int(convert to bigint by default) and target column is int type
```
ERROR 1064 (HY000): type of exprs is not match slot's, expr_type=7, slot_type=5, slot_name=F_Completion backend [id=10001] [host=172.26.92.227]
```
## What I'm doing:

- Add `castOutputColumnsTypeToTargetColumns` in LogicalPlan to OptExpression like `InsertPlanner`.

Fixes #issue

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Review by luohaha — DISMISSED** [review]

**Review by kangkaisen — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/UpdatePlanner.java [hunk 1]
```diff
@@ -170,4 +191,42 @@ public ExecPlan plan(UpdateStmt updateStmt, ConnectContext session) {
...
+        Preconditions.checkState(colNames.size() == outputColumns.size(), "Column name's size %s should be equal " +
+                        "to output column refs' size %s", colNames.size(), outputColumns.size());
+
+        for (int columnIdx = 0; columnIdx < outputColumns.size(); ++columnIdx) {
+            ColumnRefOperator outputColumn = outputColumns.get(columnIdx);
+            String colName = colNames.get(columnIdx);
+            Column column = targetTable.getColumn(colName);
+            Preconditions.checkState(column != null, "Column %s not found in table %s", colName,
+                    targetTable.getName());
+            if (!column.getType().matchesType(outputColumn.getType())) {
```
Comment (Seaven) [inline fe/fe-core/src/main/java/com/starrocks/sql/UpdatePlanner.java:221]: check the type can cast to target type？

File: fe/fe-core/src/main/java/com/starrocks/sql/UpdatePlanner.java [hunk 2]
```diff
@@ -170,4 +191,42 @@ public ExecPlan plan(UpdateStmt updateStmt, ConnectContext session) {
...
+        ScalarOperatorRewriter rewriter = new ScalarOperatorRewriter();
+        List<ScalarOperatorRewriteRule> rewriteRules = Arrays.asList(new FoldConstantsRule());
+        Preconditions.checkState(colNames.size() == outputColumns.size(), "Column name's size %s should be equal " +
+                        "to output column refs' size %s", colNames.size(), outputColumns.size());
+
+        for (int columnIdx = 0; columnIdx < outputColumns.size(); ++columnIdx) {
+            ColumnRefOperator outputColumn = outputColumns.get(columnIdx);
+            String colName = colNames.get(columnIdx);
+            Column column = targetTable.getColumn(colName);
+            Preconditions.checkState(column != null, "Column %s not found in table %s", colName,
```
Comment (packy92) [inline fe/fe-core/src/main/java/com/starrocks/sql/UpdatePlanner.java:219]: targetTable schema use lower case names.  Could the  colName be upper case?

Comment (LiShuMing) [reply fe/fe-core/src/main/java/com/starrocks/sql/UpdatePlanner.java:220]: `targetTable schema` will keep case-sensitive but compare with case-insensitive. 

Add more tests to cover this.

File: fe/fe-core/src/main/java/com/starrocks/sql/UpdatePlanner.java (hunk 1 above)
Comment (LiShuMing) [reply fe/fe-core/src/main/java/com/starrocks/sql/UpdatePlanner.java:222]: Add it. But this should not happen since it will fail in the analyzer phase.

Comment (LiShuMing) [reply fe/fe-core/src/main/java/com/starrocks/sql/UpdatePlanner.java:222]: Add more tests.

**Review by packy92 — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

**Comment by LiShuMing** [comment]
@mergify backport branch-3.1

**Pull request merged**